src/ipa_pc/mod.rs

```diff
@@ -176,26 +197,174 @@ impl<G: AffineCurve, D: Digest> InnerProductArgPC<G, D> {
+
+        // TODO: When we will move to Sponge-based construction, we will absorb the
+        //       batch_commitment only and then squeeze a new challenge
+        // Fresh random challenge x
+        let point = Self::compute_random_oracle_challenge(
+            &to_bytes![
+                batch_commitment,
+                points
+            ]
+                .unwrap(),
+        );
+
+        let mut opening_challenge_counter = 0;
+
+        // lambda
+        let mut cur_challenge = opening_challenges(opening_challenge_counter);
+        opening_challenge_counter += 1;
+
+        let mut computed_batch_v = G::ScalarField::zero();
+
+        for ((&v_i, y_i), x_i) in v_values.iter().zip(y_values).zip(points) {
+
+            computed_batch_v = computed_batch_v + &(cur_challenge * &((v_i - &y_i) / &(point - x_i)));
+
+            cur_challenge = opening_challenges(opening_challenge_counter);
+            opening_challenge_counter += 1;
+        }
+
+        let mut batch_values = vec![];
+        for commitment in commitments.iter() {
+            let value = batch_proof.batch_values
+                .get(commitment.label())
+                .ok_or(Error::MissingEvaluation {
+                    label: commitment.label().to_string(),
+                })?;
+            batch_values.push(*value);
+        }
+
+        // Reconstructed v value added to the check
+        batch_values.push(computed_batch_v);
+
+        // The commitment to h(X) polynomial added to the check
+        let mut commitments = commitments;
+        let labeled_batch_commitment = LabeledCommitment::new(
+            format!("Batch"),
+            Commitment { comm: batch_commitment.clone(), shifted_comm: None },
+            None
+        );
+        commitments.push(&labeled_batch_commitment);
+
+        let proof = &batch_proof.proof;
+        end_timer!(evals_time);
+
+        let check_time = start_timer!(|| "Succinct check batched polynomial");
+        let opening_challenge = Self::compute_random_oracle_challenge(
+            &to_bytes![
+                batch_proof.batch_values.values().collect::<Vec<&G::ScalarField>>(),
+                batch_commitment,
+                point
+            ]
+                .unwrap(),
+        );
+        let opening_challenges = |pow| opening_challenge.pow(&[pow]);
+
+        let d = vk.supported_degree();
+
+        // `log_d` is ceil(log2 (d + 1)), which is the number of steps to compute all of the challenges
+        let log_d = algebra::log2(d + 1) as usize;
+
+        if proof.l_vec.len() != proof.r_vec.len() || proof.l_vec.len() != log_d {
+            return Err(Error::IncorrectInputLength(
+                format!(
+                    "Expected proof vectors to be {:}. Instead, l_vec size is {:} and r_vec size is {:}",
+                    log_d,
+                    proof.l_vec.len(),
+                    proof.r_vec.len()
+                )
+            ));
+        }
```

Comment:
we MUST call check_degree_bounds() also, otherwise a prover can trick the verifier.

Reply:
Degree bound check was added to the succinct_check function